File: frontend/src/types/release.ts
export interface Release {
  id: number
  name: string
  version: string
  description: string
  status: 'planned' | 'in_progress' | 'released' | 'cancelled'
  release_date: string | null
  project_id: number
  created_at: string
  updated_at: string
  progress?: number
}

export interface ReleaseProgress {
  id: number
  name: string
  total_issues: number
  done_issues: number
  progress: number
}

export interface ReleaseCreateRequest {
  name: string
  version: string
  description?: string
  status?: string
  release_date?: string
}

export interface ReleaseUpdateRequest {
  name?: string
  version?: string
  description?: string
  status?: string
  release_date?: string
}

export interface ReleaseIssueRequest {
  issue_ids: number[]
}

File: backend/internal/dto/response/release.go
package response

type ReleaseResponse struct {
	ID          uint64  `json:"id"`
	Name        string  `json:"name"`
	Version     string  `json:"version"`
	Description string  `json:"description"`
	Status      string  `json:"status"`
	ReleaseDate *string `json:"release_date"`
	ProjectID   uint64  `json:"project_id"`
	CreatedAt   string  `json:"created_at"`
	UpdatedAt   string  `json:"updated_at"`
}

type ReleaseProgressResponse struct {
	ID          uint64 `json:"id"`
	Name        string `json:"name"`
	TotalIssues int    `json:"total_issues"`
	DoneIssues  int    `json:"done_issues"`
	Progress    int    `json:"progress"`
}